data-prepper-plugins/common/src/test/java/org/opensearch/dataprepper/plugins/s3keyindex/S3ObjectIndexTest.java

```diff
@@ -0,0 +1,116 @@
+        assertTrue(actualIndex.contains(expectedIndex));
+    }
+
+    @Test
+    void testObjectTimePattern_Exceptional_time_TooGranular() throws IllegalArgumentException {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndex.getDatePatternFormatter("events-%{yyyy-AA-dd}");
+        });
+    }
+
+    @Test
+    void testObjectTimePatterns_equal() throws IllegalArgumentException {
+
+        DateTimeFormatter expectedIndex = S3ObjectIndex.getDatePatternFormatter("events-%{yyyy-MM-dd}");
+        DateTimeFormatter actualIndex = S3ObjectIndex.getDatePatternFormatter("events-%{yyyy-MM-dd}");
+        assertEquals(actualIndex.toString(), expectedIndex.toString());
+    }
+
+    @Test
+    void test_utc_current_time() throws IllegalArgumentException {
+
+        ZonedDateTime expectedIndex = S3ObjectIndex.getCurrentUtcTime();
+        ZonedDateTime actualIndex = S3ObjectIndex.getCurrentUtcTime();
+
+        assertEquals(expectedIndex.getDayOfYear(), actualIndex.getDayOfYear());
+        assertEquals(expectedIndex.getDayOfMonth(), actualIndex.getDayOfMonth());
+        assertEquals(expectedIndex.getDayOfWeek(), actualIndex.getDayOfWeek());
+    }
+
+    @Test
+    void testObjectTimePattern_Exceptional_TooGranular() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndex.getDatePatternFormatter("events-%{yyyy-AA-ddThh:mm}");
+        });
+    }
+
+    @Test
+    void testObjectTimePattern_Exceptional_at_theEnd() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndex.getDatePatternFormatter("events-%{yyy{MM}dd}");
+        });
+    }
+
+    @Test
+    void testObject_allows_one_date_time_pattern_Exceptional() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndex.getDatePatternFormatter("events-%{yyyy-MM-dd}-%{yyyy-MM-dd}");
+        });
+    }
+
+    @Test
+    void testObject_nested_pattern_Exceptional() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndex.getDatePatternFormatter("bucket-name-\\%{\\%{yyyy.MM.dd}}");
+        });
+    }
+
+    @Test
+    void testObject_null_time_pattern() throws NullPointerException {
+        assertNull(S3ObjectIndex.getDatePatternFormatter("bucket-name"));
+    }
+
+    @Test
+    void testObjectAliasWithDatePrefix_Exceptional_time_TooGranular() throws IllegalArgumentException {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndex.getObjectNameWithDateTimeId("events-%{yyyy-AA-dd}");
+        });
+    }
+
+    @Test
+    void testObjectAliasWithDatePrefix_equal() throws IllegalArgumentException {
+
+        String expectedIndex = S3ObjectIndex.getObjectNameWithDateTimeId("events-%{yyyy-MM-dd}");
+        String actualIndex = S3ObjectIndex.getObjectNameWithDateTimeId("events-%{yyyy-MM-dd}");
+        assertNotEquals(actualIndex.toString(), expectedIndex.toString());

```

Comment:
Why is this assertNotEquals()?

Reply:
As per requirement, EPOCH SECONDS and UniqueId are appended for the name_pattern. 
assertNotEquals are added  since both the responses are not equal.

Reply:
You mean it would be expected String + EPOCH seconds + unique id? In that case, can we compare that expected String is in the `actualIndex.toString()`?